resource_customizations/gateway.solo.io/VirtualService/health.lua

```diff
@@ -0,0 +1,27 @@
+if not obj.status then
+  return {status="Progressing", message="Empty status field"}
+end
+
+states={[0]=0, [1]=0, [2]=0, [3]=0}
+messages={}
+if obj.status.state then
+  states[obj.status.state] = 1
+  messages[obj.status.state] = obj.status.reason or "Unknown"
+elseif obj.status.statuses then
+  for k,v in pairs(obj.status.statuses or {}) do
+    states[v.state] = 1
+    messages[v.state] = v.reason or "Unknown"
+  end
```

Comment:
If I'm reading this correctly, only the last status's state will be reported. Is that intended?